### ISSUE TITLE
Unsafe any in catch

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3010]
### Bug Report

- __TSLint version__: 4.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: gulp

#### TypeScript code being linted

```ts
try {} catch (e) { 
  var a = e.toString();
}
```

with `tslint.json` configuration:

```json
no-unsafe-any: true
```

#### Actual behavior
error noUnsafeAny: Unsafe use of expression of type 'any'.

#### Expected behavior
No error. It is impossible to use try-catch with the rule even for logging error message.

**Comment by NN---** [comment]
e can be null/undefined too.

**Comment by pburrows** [comment]
So what is the solution? I have been adding
`// tslint:disable-next-line:no-unsafe-any`
in my catch clauses. But is there a better way?

**Comment by NN---** [comment]
Use template string:
```ts
try { throw null; } catch (e) { console.log(`${e} thrown`) }
```

**Comment by tommedema** [comment]
@NN--- That's hardly a solution... E.g. my case:

I ended up using @AndyJ's comment:

    /**
     * Retrieves a component template from filesystem
     */
    const getComponentTemplate = async (
      p: string
    ): Promise<string> => {
      let template: string
      try {
        template = await fs.readFile(p, {
          encoding: 'utf8'
        })
      } catch (e) {
        if (isNodeError(e) && e.code === 'ENOENT') {
          throw new Error(`template for element type ${elementType} not found`)
        }
        throw e
      }
    
      return template
    }
    
    /**
     * @param error the error object.
     * @returns if given error object is a NodeJS error.
     */
    const isNodeError = (error: Error): error is NodeJS.ErrnoException =>
      error instanceof Error

TSLint complains `Unsafe use of expression of type 'any'.`

Why would I want to use a template string here?

**Comment by zquancai** [comment]
/* tslint:disable:no-unsafe-any */
// your code
/* tslint:disable:no-unsafe-any */

**Comment by NN---** [comment]
Change isNodeError to receive any or better unknown if you have TS 3.0.
```ts
const isNodeError = (error: any): error is NodeJS.ErrnoException =>
  error instanceof Error;
```
